### ISSUE TITLE
Remove Hidden Keywords Meta Issue

### ISSUE THREAD

**Issue by Hanmac** [Card-Forge/forge#3307]
This is a Meta Issue for all the others refering Hidden Keywords that needs to be refactored, either as ReplacementEffect=CantHappen or as Static Ability that doesn't use Layer

* IgnoreLandfall #2507 
* `CARDNAME can't be regenerated` #3304 
* Block as If Shadow #2550
* Can't Untap #984
* `CARDNAME's power and toughness are switched` , this probably can be done as flags instead of keyword? should also use Layer?
*  `All creatures able to block CARDNAME do so.` & `CARDNAME must be blocked if able.`
* `CARDNAME can't be countered.`  #2688
* `CARDNAME assigns no combat damage`
* `Colorless Damage Source` 
* `CARDNAME can't attack or block.`
* `CARDNAME's activated abilities can't be activated.`

Both `CARDNAME can't attack or block.` and  `CARDNAME's activated abilities can't be activated.` are tricky because of
`Volrath's Curse` and `Lost in Thought` having `ignore this effect` on them, so they either needs to be one static ability, or some trickery

**Comment by Hanmac** [comment]
currently `Heart of Kiran` isn't directly a hidden keyword, but its ability still should be reworked into a better way to replace the cost

see the other cost replacer:

* "CyclingForZero" (Activation)
* "You may pay 0 rather than pay equip costs." (Activation)
* "You may pay 0 rather than pay the echo cost for permanents you control." (Effect)

**Comment by tool4ever** [comment]
After all are finished the logic for `isStackingKeyword` can also be changed to use Keyword class information instead of ugly `NonStackingKWList.txt`

**Comment by Hanmac** [comment]
@tool4ever i already started this with `KeywordInterface.redundant`

also related to CardTextChanges, i want the keyword teach about what part of the Keyword should be affected by Text Changes, so it would simplify this issue: #2512


If you want you can try to update the CantUntap stuff again (using CantHappen Replacement?)

**Comment by tool4ever** [comment]
about IgnoreEffectCost$:
the other 2 cards that use it to affect Players wouldn't really need to be done as Continuous either
- so the logic for it could be moved to work on generic StaticAbility instead
- to avoid some redundant code the CantSearchLibrary stuff should also be Static instead of RE
- this would allow cleaning up adding statics on Players (only used by _Damping Engine_ it seems)
  - to make this work we might consider supporting more than one mode together, e.g. `Mode$ CantPlayLand, CantBeCast`

**Comment by Hanmac** [comment]
Yeah, the IgnoreCost is somewhat tricky, but I would like to see you try it

For the enchant creature ones,  doing a Static that gives itself another Statics might not big of a problem 

For the Player ones, yeah, I recently stumbled upon the Player addStatic thing only used for Damping Engine and I would happily remove it

**Comment by Hanmac** [comment]
@tool4ever can you help me take a look at _Undiscovered Paradise_?
> During your next untap step, as you untap your permanents, return CARDNAME to its owner's hand.

Shouldn't this be able to be done as DelayedTrigger? Or something?

**Comment by Hanmac** [comment]
Removed note for `Volrath's Curse` and similar

`CARDNAME can't attack or block.` and `CARDNAME's activated abilities can't be activated.` are mostly left for Detain like Effects. See #7209

**Comment by Hanmac** [comment]
I really want to remove `HIDDEN CARDNAME can't block.`, but for this, I need to make Detain merged first
